### ISSUE TITLE
Consistent and safe divide-by-zero handling

### ISSUE THREAD

**Issue by stuart-cls** [Quasars/orange-spectroscopy#302]
In several places we have `/` or `/=` operators (at least Normalize, NormalizeReference, Absorbance, Transmittance).

Currently, we have:
`0 / 0 == Nan` -> Okay, as Orange can handle NaNs
`1 / 0 == inf` -> Not okay, as Orange / OWHyper chokes on (+/-) infs
`-1 / 0 == -inf` -> ditto

We could do something like `X[np.isinf(X)] = np.nan` after each divide or in a helper function? 

Or we could not re-invent the wheel and use this from scikit-learn:

```
def _handle_zeros_in_scale(scale, copy=True):
    ''' Makes sure that whenever scale is zero, we handle it correctly.
    This happens in most scalers when we have constant features.'''

    # if we are fitting on 1D arrays, scale might be a scalar
    if np.isscalar(scale):
        if scale == .0:
            scale = 1.
        return scale
    elif isinstance(scale, np.ndarray):
        if copy:
            # New array to avoid side-effects
            scale = scale.copy()
        scale[scale == 0.0] = 1.0
return scale
```

from https://github.com/scikit-learn/scikit-learn/blob/7b136e92acf49d46251479b75c88cba632de1937/sklearn/preprocessing/data.py#L70

This **changes the behaviour to keep val / 0 as val** but makes Normalize.Vector and the rest consistent, since Vector uses the sci-kit code already.

@markotoplak Thoughts?
Anyone else have strong opinions on divide-by-zero handling?

**Comment by markotoplak** [comment]
I dislike sklearn's approach that quietly ignores the operation. I'd prefer to keep such values NaN. Do you see any advantages for this approach?

I like your utility function idea. And for consistency, I would reimplement Normalize.Vector. :)

**Comment by stuart-cls** [comment]
I agree, it is unexpected to get an un-scaled value in the middle of a scaled plot. 

Let's go with:

- [ ] helper function for consistent behaviour which returns NaN for all divzero cases
- [ ] re-implement Normalize.Vector to match

as you suggest.

If anyone (myself included) starts working on this, just assign to yourself :)

**Comment by stuart-cls** [comment]
To clarify the status of this issue: `Normalize.Vector` still uses sklearn's code and silently ignores the normalization so it should be re-implemented to not do that.
